selftest/ensure_equals.cpp

```diff
@@ -379,5 +379,24 @@ void object::test<20>()
     }
 }
 
+/**
+* Checks positive ensure_equals with mixing floating point types (float + double)
+*/
+template<>
+template<>
+void object::test<21>()
+{
+#if __cplusplus < 201103L
+    skip();
+#endif
```

Comment:
If we don't use the new template function, compiler will resolve `double` + `float` to the most generic function because there is no conversion to do. Thus, it won't use the `double` + `double` version of `ensure_equals` (because of the conversion `float` to `double`), and it will try to compare the values like an `int` with a simple `!=`.

Thus the `skip()` if we can't use the new template function.